rust/src/smb/smb_status.rs

```diff
@@ -0,0 +1,3609 @@
+pub const SMB_NTSTATUS_NDIS_DOT11_POWER_STATE_INVALID:                                 u32 = 0xc0232002;
+pub const SMB_NTSTATUS_NDIS_PM_WOL_PATTERN_LIST_FULL:                                  u32 = 0xc0232003;
+pub const SMB_NTSTATUS_NDIS_PM_PROTOCOL_OFFLOAD_LIST_FULL:                             u32 = 0xc0232004;
+pub const SMB_NTSTATUS_IPSEC_BAD_SPI:                                                  u32 = 0xc0360001;
+pub const SMB_NTSTATUS_IPSEC_SA_LIFETIME_EXPIRED:                                      u32 = 0xc0360002;
+pub const SMB_NTSTATUS_IPSEC_WRONG_SA:                                                 u32 = 0xc0360003;
+pub const SMB_NTSTATUS_IPSEC_REPLAY_CHECK_FAILED:                                      u32 = 0xc0360004;
+pub const SMB_NTSTATUS_IPSEC_INVALID_PACKET:                                           u32 = 0xc0360005;
+pub const SMB_NTSTATUS_IPSEC_INTEGRITY_CHECK_FAILED:                                   u32 = 0xc0360006;
+pub const SMB_NTSTATUS_IPSEC_CLEAR_TEXT_DROP:                                          u32 = 0xc0360007;
+pub const SMB_NTSTATUS_IPSEC_AUTH_FIREWALL_DROP:                                       u32 = 0xc0360008;
+pub const SMB_NTSTATUS_IPSEC_THROTTLE_DROP:                                            u32 = 0xc0360009;
+pub const SMB_NTSTATUS_IPSEC_DOSP_BLOCK:                                               u32 = 0xc0368000;
+pub const SMB_NTSTATUS_IPSEC_DOSP_RECEIVED_MULTICAST:                                  u32 = 0xc0368001;
+pub const SMB_NTSTATUS_IPSEC_DOSP_INVALID_PACKET:                                      u32 = 0xc0368002;
+pub const SMB_NTSTATUS_IPSEC_DOSP_STATE_LOOKUP_FAILED:                                 u32 = 0xc0368003;
+pub const SMB_NTSTATUS_IPSEC_DOSP_MAX_ENTRIES:                                         u32 = 0xc0368004;
+pub const SMB_NTSTATUS_IPSEC_DOSP_KEYMOD_NOT_ALLOWED:                                  u32 = 0xc0368005;
+pub const SMB_NTSTATUS_IPSEC_DOSP_MAX_PER_IP_RATELIMIT_QUEUES:                         u32 = 0xc0368006;
+pub const SMB_NTSTATUS_VOLMGR_MIRROR_NOT_SUPPORTED:                                    u32 = 0xc038005b;
+pub const SMB_NTSTATUS_VOLMGR_RAID5_NOT_SUPPORTED:                                     u32 = 0xc038005c;
+pub const SMB_NTSTATUS_VIRTDISK_PROVIDER_NOT_FOUND:                                    u32 = 0xc03a0014;
+pub const SMB_NTSTATUS_VIRTDISK_NOT_VIRTUAL_DISK:                                      u32 = 0xc03a0015;
+pub const SMB_NTSTATUS_VHD_PARENT_VHD_ACCESS_DENIED:                                   u32 = 0xc03a0016;
+pub const SMB_NTSTATUS_VHD_CHILD_PARENT_SIZE_MISMATCH:                                 u32 = 0xc03a0017;
+pub const SMB_NTSTATUS_VHD_DIFFERENCING_CHAIN_CYCLE_DETECTED:                          u32 = 0xc03a0018;
+pub const SMB_NTSTATUS_VHD_DIFFERENCING_CHAIN_ERROR_IN_PARENT:                         u32 = 0xc03a0019;
+
+pub fn smb_ntstatus_string(c: u32) -> String {
```

Comment:
I'm a bit worried about a possible performance overhead of a function like this. I don't know how Rust implements this, but if it is a linear lookup it might be very expensive. We use this in the packet path for smb logging.

@jasonish any ideas?

Reply:
It looks like Rust or LLVM optimizes this by turning it into a jump list and should perform better than the alternative which would most likely be a hashmap.

Reply:
I'm all for such improvements, esp since patterns like this tend to get copied into new code.

Reply:
Done